src/ham/hamd/hamd.cpp

```diff
@@ -365,6 +366,40 @@ ::DBus::Struct< bool, std::string > hamd_c::set_roles(const std::string& login,
     return ret;
 }
 
+/**
+ * @brief Create a new user or modify an existing user
+ *
+ * @param login     User's login name
+ * @param roles     List of roles
+ * @param hashed_pw Hashed password. Must follow usermod's --password
+ *                  syntax.
+ */
+::DBus::Struct< bool, std::string > hamd_c::usermod(const std::string                & login,
+                                                    const std::vector< std::string > & roles,
+                                                    const std::string                & hashed_pw)
+{
+    ::DBus::Struct< bool,       /* success */
+                    std::string /* errmsg */   > ret;
+    struct passwd * pwd = ::getpwnam(login.c_str());
+    if (pwd == nullptr)
+    {   // Add user if it doesn't exist
+    	return useradd(login,roles,hashed_pw);
+
+    }
+    else // User exists so update password and role
+    {
+        ret = passwd(login,hashed_pw);
+        if (ret._1 == true)
```

Comment:
You don;t need to check a boolean againts true/false. You can just say `if (ret._1)`

Reply:
I've addressed all your comments as well as added a spytest. Thank you.